### DISCUSSION TITLE
How to request the confirmation before executing a command?

### DISCUSSION THREAD

**Discussion by dstd** [mitmproxy/mitmproxy#7282] [Q&A] [answered]
There are some number of destructive commands, such as `view.flows.remove`, that I would like to have a confirmation request before executing. Especially considering how easy it is to make a mistake between `z` and `shift+z`. Is there any way to implement this?

I did some sort of workaround where pressing `d` has to be confirmed by pressing `Enter`, and confirmless version moved to `ctrl+d`:
```
-
  ctx: ["flowlist","flowview"]
  key: d
  cmd: console.command view.flows.remove @focus
-
  ctx: ["flowlist","flowview"]
  key: ctrl d
  cmd: view.flows.remove @focus
-
  ctx: ["flowlist"]
  key: z
  cmd: console.command view.flows.remove @all
```

IMHO this is not quite a good way. Ideally, I'd like to do it like this:
```
-
  ctx: ["flowlist","flowview"]
  key: d
  cmd: console.status_prompt_for_command "Delete focused" view.flows.remove @focus
-
  ctx: ["flowlist"]
  key: z
  cmd: console.status_prompt_for_command "Clear flow list" view.flows.remove @all
```
So that it will prompt a confirmation like:
```
Clear flow list (yes,no)? ▉
```

**Comment by Prinzhorn** [comment, marked as answer]
This has just been implemented and will be in the next release https://github.com/mitmproxy/mitmproxy/pull/7445
